File: lib-esphome-client/src/test/java/io/github/arhor/esphome/client/async/EspHomeClientConfigTest.java
package io.github.arhor.esphome.client.async;

import org.junit.jupiter.api.Test;

import java.time.Duration;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertThrows;

class EspHomeClientConfigTest {

    @Test
    void encryptionEnabledRequiresKey() {
        assertThrows(IllegalArgumentException.class, () -> new EspHomeClient.Config.Encryption(true, null));
    }

    @Test
    void defaultEncryptionDisabled() {
        final var config = new EspHomeClient.Config("host", 6053, "client", null);
        assertFalse(config.encryption().enabled());
    }

    @Test
    void rejectsZeroConnectTimeout() {
        assertThrows(
            IllegalArgumentException.class,
            () -> new EspHomeClient.Config(
                "host",
                6053,
                "client",
                null,
                EspHomeClient.Config.Encryption.disabled(),
                Duration.ZERO,
                Duration.ofSeconds(5),
                EspHomeClient.API_VERSION_MAJOR,
                EspHomeClient.API_VERSION_MINOR
            )
        );
    }
}

File: services/lib-esphome-client/src/test/kotlin/io/github/arhor/esphome/client/EspHomeClientConfigTest.kt
package io.github.arhor.esphome.client

import io.github.arhor.esphome.client.config.EspHomeClientConfig
import io.github.arhor.esphome.client.config.EspHomeEncryptionConfig
import java.time.Duration
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFailsWith

class EspHomeClientConfigTest {

    @Test
    fun `uses safe defaults for plaintext native api`() {
        val config = EspHomeClientConfig(host = "esp32-cam.local")

        assertEquals(6053, config.port)
        assertEquals("hive-lib-esphome-client", config.clientName)
        assertEquals(Duration.ofSeconds(2), config.connectTimeout)
        assertEquals(Duration.ofSeconds(5), config.readTimeout)
        assertEquals(null, config.password)
        assertEquals(EspHomeEncryptionConfig(), config.encryption)
    }

    @Test
    fun `rejects invalid endpoint and timeout values`() {
        assertFailsWith<IllegalArgumentException> { EspHomeClientConfig(host = " ") }
        assertFailsWith<IllegalArgumentException> { EspHomeClientConfig(host = "camera", port = 0) }
        assertFailsWith<IllegalArgumentException> {
            EspHomeClientConfig(host = "camera", connectTimeout = Duration.ZERO)
        }
        assertFailsWith<IllegalArgumentException> {
            EspHomeClientConfig(host = "camera", readTimeout = Duration.ZERO)
        }
    }

    @Test
    fun `rejects enabled encryption without key`() {
        assertFailsWith<IllegalArgumentException> {
            EspHomeEncryptionConfig(enabled = true, key = null)
        }
        assertFailsWith<IllegalArgumentException> {
            EspHomeEncryptionConfig(enabled = true, key = " ")
        }
    }
}
